src/stories/form-ui.tsx

```diff
@@ -0,0 +1,94 @@
+import { FormEventHandler, useState } from "react";
+
+export default function FormUI() {
+    const [passwordError, setPasswordError] = useState<string | undefined>();
+    const [confirmPasswordError, setConfirmPasswordError] = useState<
+        string | undefined
+    >();
+
+    const handleSubmit: FormEventHandler<HTMLFormElement> = (formInput) => {
+        formInput.preventDefault();
+        const fullname = formInput.currentTarget.fullname.value;
+        const email = formInput.currentTarget.email.value;
+        const password: string = formInput.currentTarget.password.value;
+        const confirmPassword =
+            formInput.currentTarget["confirm-password"].value;
+
+        if (password.length < 8) {
+            setPasswordError("Password must be at least 8 charakters long.");
+        } else if (password !== confirmPassword) {
```

Comment:
Having this in an `else if` means there can only ever be one error at a time, which doesn't really make sense.